### PR TITLE
Suggestion

### PR DESCRIPTION
**Pull request by snacktavish**

### CODE REVIEW

File: modules/vcffixer.py [hunk 1]
```diff
@@ -103,87 +98,79 @@ def process_vcf(vcf_file, dict):
...
-                        align_list[num] = selected_alt.upper()
-                        print("N replaced with {} and position {} from vcf position {}.".format(selected_alt, num, pos_in_vcf))
-
-        # elif nuc.upper() in nuc_set_:
-        #     output_list.append(nuc.upper())
-
-        elif nuc.upper() != "N":
-            if nuc.upper() not in nuc_set_:
-            # output_list.append("N")
-                align_list[num] = "N"
```
Comment (snacktavish) [inline modules/vcffixer.py:151]: This is where it was removing degenerate nucleotides - and we can put that back in! But we need to be specific about the choice.

File: modules/vcffixer.py [hunk 2]
```diff
@@ -103,87 +98,79 @@ def process_vcf(vcf_file, dict):
...
-            # ref_and_alt = vcf_dict[pos_in_vcf]
             ref_and_alt = vcf_dict.get(pos_in_vcf)
             if ref_and_alt == None:
-                output_list.append("N")
+                output_list.append(nuc)
             elif ref_and_alt != None:
                 ref = ref_and_alt[0]
-                alt = ref_and_alt[1]
-                split_alts = alt.split(',')
-                if len(split_alts) > 1:
```
Comment (snacktavish) [inline modules/vcffixer.py:134]: What was happening if there was only one alternative base? There is no else:

**Comment by jtfield** [comment]
Rejected until the source of off-by-one errors are found and fixed.

**Comment by snacktavish** [comment]
Passes tests/comprehensive_test.sh and the new pytest tests on main.

**Pull request merged**